log/configmanager.go

```diff
@@ -1,238 +0,0 @@
-	return [...]string{"loglevel", "kafka"}[c]
-}
-
-type ChangeEventType int
-
-const (
-	Put ChangeEventType = iota
-	Delete
-)
-
-func (c ChangeEventType) EventString() string {
-	return [...]string{"Put", "Delete"}[c]
-}
-
-type ConfigChangeEvent struct {
-	ChangeType int
-	Key        string
-	Value      interface{}
-}
-
-type ConfigManager struct {
-	backend *model.Backend
-}
-
-//Create New ConfigManager
-func NewConfigManager(kvClient kvstore.Client, kvStoreType, kvStoreHost, kvStoreDataPrefix string, kvStorePort, kvStoreTimeout int) *ConfigManager {
-	// Setup the KV store
-	var cm ConfigManager
-	cm.backend = &model.Backend{
-		Client:     kvClient,
-		StoreType:  kvStoreType,
-		Host:       kvStoreHost,
-		Port:       kvStorePort,
-		Timeout:    kvStoreTimeout,
-		PathPrefix: kvStoreDataPrefix}
-	return &cm
-}
-
-//populateLogLevel store the loglevel retrieved from kvstore to LogLevel
-func populateLogLevel(data map[string]interface{}) []LogLevel {
-	loglevel := []LogLevel{}
-	for _, v := range data {
-		switch val := v.(type) {
-		case []byte:
-			json.Unmarshal(val, &loglevel)
-		}
-	}
-
-	return loglevel
-}
-
-//Initialize the component config
-func InitComponentConfig(componentLabel string, configType ConfigType, cm *ConfigManager) (*ComponentConfig, error) {
-	//construct ComponentConfig
-	//verify componentConfig for provided componentLabel and configType is present in etcd,
-
-	cConfig := &ComponentConfig{
-		componentLabel:   componentLabel,
-		configType:       configType,
-		CManager:         cm,
-		changeEventChan:  nil,
-		kvStoreEventChan: nil,
-		logLevel:         nil,
-	}
-
-	return cConfig, nil
-}
-
-type ComponentConfig struct {
-	CManager         *ConfigManager
-	componentLabel   string
-	configType       ConfigType
-	changeEventChan  chan *ConfigChangeEvent
-	kvStoreEventChan chan *kvstore.Event
-	logLevel         []LogLevel
-}
-
-type LogLevel struct {
-	PackageName string
-	Level       string
-}
-
-//MonitorForConfigChange watch on the keys
-//If any changes happen then process the event create new ConfigChangeEvent and return
-func (c *ComponentConfig) MonitorForConfigChange() (chan *ConfigChangeEvent, error) {
-	//call makeConfigPath function to create path
-	key := c.makeConfigPath()
-
-	//call backend createwatch method
-	c.kvStoreEventChan = make(chan *kvstore.Event)
-	c.changeEventChan = make(chan *ConfigChangeEvent)
-
-	c.kvStoreEventChan = c.CManager.backend.CreateWatch(key)
-
-	go c.processKVStoreWatchEvents(key)
-
-	return c.changeEventChan, nil
-}
-
-//processKVStoreWatchEvents process the kvStoreEventChan and create ConfigChangeEvent
-func (c *ComponentConfig) processKVStoreWatchEvents(key string) {
-	//In a loop process incoming kvstore events and push changes to changeEventChan
-
-	//call defer backend delete watch method
-
-	for watchResp := range c.kvStoreEventChan {
-		configEvent := newChangeEvent(watchResp.EventType, watchResp.Key, watchResp.Value)
-		c.changeEventChan <- configEvent
```

Comment:
You should be removing the complete path here; before passing on the event. 

Also, a validation can be done to verify that event belongs to this ComponentConfig. Call makeConfigPath() to get prefix path, verify that the key in Event has that as prefix and then strip it before constructing ConfigChangeEvent